homeassistant/components/ecodevices/config_flow.py

```diff
@@ -0,0 +1,164 @@
+"""Config flow to configure the GCE Eco-Devices integration."""
+from pyecodevices import (
+    EcoDevices,
+    EcoDevicesCannotConnectError,
+    EcoDevicesInvalidAuthError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import DEVICE_CLASSES as SENSOR_DEVICE_CLASSES
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    CONF_C1_DEVICE_CLASS,
+    CONF_C1_ENABLED,
+    CONF_C1_NAME,
+    CONF_C1_UNIT_OF_MEASUREMENT,
+    CONF_C2_DEVICE_CLASS,
+    CONF_C2_ENABLED,
+    CONF_C2_NAME,
+    CONF_C2_UNIT_OF_MEASUREMENT,
+    CONF_T1_ENABLED,
+    CONF_T1_NAME,
+    CONF_T1_UNIT_OF_MEASUREMENT,
+    CONF_T2_ENABLED,
+    CONF_T2_NAME,
+    CONF_T2_UNIT_OF_MEASUREMENT,
+    DEFAULT_C1_NAME,
+    DEFAULT_C2_NAME,
+    DEFAULT_T1_NAME,
+    DEFAULT_T1_UNIT_OF_MEASUREMENT,
+    DEFAULT_T2_NAME,
+    DEFAULT_T2_UNIT_OF_MEASUREMENT,
+    DOMAIN,
+)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=80): int,
+        vol.Required(CONF_SCAN_INTERVAL, default=5): int,
```

Comment:
If a user needs to configure this, it should be part of the OptionsFlow imo. https://developers.home-assistant.io/docs/config_entries_options_flow_handler/#options-support

Are there use cases where a user needs / wants to configure this?